src/components/DocumentationTopic/DocumentationNav/VersionToggle.vue

```diff
@@ -0,0 +1,283 @@
+      // for the browser orientation-change transitions to finish
+      await waitFrames(3);
+      this.calculateSelectWidth();
+    }, 150, true);
+    window.addEventListener('resize', cb);
+    window.addEventListener('orientationchange', cb);
+    this.$once('hook:beforeDestroy', () => {
+      window.removeEventListener('resize', cb);
+      window.removeEventListener('orientationchange', cb);
+    });
+  },
+  updated() {
+    this.versionModel = this.currentVersion;
+  },
+  watch: {
+    // interfaceversion: {
+    //   immediate: true,
+    //   handler(version) {
+    //     this.versionModel = version;
+    //   },
+    // },
+    versionModel: {
+      immediate: true,
+      handler: 'calculateSelectWidth',
+    },
+  },
+  methods: {
+    /**
+     * Returns a formatted route object
+     * @param {{ query: string, path: string }} route - a config object passed by the render JSON
+     * @returns {{ path: (null|string), query: { version: (string|undefined) }}}
+     */
+    getRoute(route) {
+      // pass undefined to remove the query param if its most recent version
+      const version = route.query === this.versionList[0] ? undefined : route.query;
+      return {
+        // make sure we dont loose any extra query params on the way
+        query: { ...this.$route.query, version },
+        path: this.isCurrentPath(route.path) ? null : this.normalizePath(route.path),
+      };
+    },
+    // foo(version) {
+    //   this.store.setPreferredVersion(version);
+    // },
+    pushRoute(route) {
+      // Persist the selected version as a preference in the store (backed by
+      // the browser's local storage so that it can be retrieved later for
+      // subsequent navigation without the query parameter present)
+      this.store.setPreferredVersion(route.query);
+      // Navigate to the version variant page
+      this.$router.push(this.getRoute(route));
+    },
+    isCurrentPath(path) {
+      // the `.replace` call is needed since paths vended by the backend do not
+      // include a leading slash, while the router provided path does
+      return this.$route.path.replace(/^\//, '') === path;
+    },
+    normalizePath(path) {
+      // Sometimes `paths` data from `variants` are prefixed with a leading
+      // slash and sometimes they aren't
+      return path.startsWith('/') ? path : `/${path}`;
+    },
+    /**
+     * Calculated the width of the select by fetching it from the faux select.
+     * @return {Promise<void>}
+     */
+    async calculateSelectWidth() {
+      // await next tick, so we are sure the element is rendered.
+      await this.$nextTick();
+      this.adjustedWidth = this.$refs['version-sizer'].clientWidth + 6;
+    },
+  },
+  computed: {
+    singleVersionPage() {
+      return (this.versionList && this.versionList.length === 1) ? this.versionList[0] : null;
+    },
+    manyVersions() {
+      return !this.singleVersionPage && this.versionList;
```

Comment:
See my other comment, if `versionList` is not `null`, then this is probably not necessary.

Reply:
I suggest replacing this computed property with

```js
hasMultipleVersions: ({ versionList }) => versionList.length > 1,
```

you should be able to use it in line 29 too.
No need to check if this.versionList exists because it will always do as an empty array